core/sds-aspect-model-java-generator/src/test-shared/java/io/openmanufacturing/sds/test/shared/compiler/JavaCompiler.java

```diff
@@ -45,19 +56,45 @@ public static Map<QualifiedName, Class<?>> compile( final List<QualifiedName> lo
             .map( key -> new CompilerInput( key.toString(), sources.get( key ) ) )
             .collect( Collectors.toList() );
 
+      if ( System.getProperty( WRITE_SOURCES_PROPERTY ) != null ) {
```

Comment:
Why did you make this optional? To me it sound really helpful to have the generated classes from the tests written to an explicit directory.

Reply:
Yes, I would like to have this as well. The problem is that many model test models define model elements with the same name (e.g. "TestEntity") but with different structure and semantics. When you selectively run one test, this is no problem. When you run the test suite, where code for all test models is generated, you end up in a situation where one test overwrites previously generated code from another test.
Let's say AspectA has a TestEntity and AspectB also has TestEntity but with a different structure. Then AspectB's TestEntity's constructor does not match the `new TestEntity` cann in AspectA and you end up with compilation errors. We would have to cleanly move each test model into its own namespace. I consider this out of scope for this PR though.